libcxx/test/std/atomics/atomics.ref/notify_all.pass.cpp

```diff
@@ -6,6 +6,7 @@
 //===----------------------------------------------------------------------===//
 
 // UNSUPPORTED: c++03, c++11, c++14, c++17
+// UNSUPPORTED: no-threads
```

Comment:
What occurs when you run these tests at the moment?

Do they hang forever? Do we expect them to fail (and should consider `XFAIL`)?

Reply:
Does not compile
https://github.com/llvm/llvm-project/actions/runs/8939256898/job/24556637589#step:3:1864

I am following what is done in the `std::atomic` tests
https://github.com/llvm/llvm-project/blob/5d81b1c50a6eadc5d2ebe8c524dd2daa665402d7/libcxx/test/std/atomics/atomics.types.generic/atomics.types.float/notify_all.pass.cpp#L8

Reply:
This test and several others fail on apple https://buildkite.com/llvm-project/libcxx-ci/builds/35217#_. These platforms lack synchronization. This should fix these builds.
```suggestion
// XFAIL: availability-synchronization_library-missing

```